packages/cloudflare/src/entrypoints/image-service.ts

```diff
@@ -0,0 +1,43 @@
+import type { ExternalImageService } from 'astro';
+
+import { baseService } from 'astro/assets';
+import { isESMImportedImage, isRemoteAllowed, joinPaths } from '../utils/assets.js';
+
+const service: ExternalImageService = {
+	validateOptions: baseService.validateOptions,
+	getHTMLAttributes: baseService.getHTMLAttributes,
+	getSrcSet: baseService.getSrcSet,
+	getURL: (options, imageConfig) => {
+		const resizingParams = [];
+		if (options.width) resizingParams.push(`width=${options.width}`);
+		if (options.height) resizingParams.push(`height=${options.height}`);
+		if (options.quality) resizingParams.push(`quality=${options.quality}`);
+		if (options.fit) resizingParams.push(`fit=${options.fit}`);
+		if (options.format) resizingParams.push(`format=${options.format}`);
+
+		let imageSource = '';
+		if (isESMImportedImage(options.src)) {
+			imageSource = options.src.src;
+		} else if (isRemoteAllowed(options.src, imageConfig)) {
+			imageSource = options.src;
+		} else {
+			// If it's not an imported image, nor is it allowed using the current domains or remote patterns, we'll just return the original URL
+			return options.src;
+		}
+
+		const imageEndpoint = joinPaths(
+			// FIXME: I have no idea why we get:
+			// Property 'env' does not exist on type 'ImportMeta'.ts(2339)
+
+			// @ts-ignore
+			import.meta.env.BASE_URL,
```

Comment:
Anyone has an idea, how to fix the type error?

Reply:
You could `import "astro/client"`, but the downside is that those types leak out into the entire project.

Reply:
I wonder if it is worth, or if we just keep the `// @⁠ts-ignore`

Reply:
Yeah, it seems like a good place for it. `ignore` or `expect-error`.